djoser/serializers.py

```diff
@@ -24,6 +24,16 @@ class Meta:
             User.USERNAME_FIELD,
         )
 
+    def update(self, instance, validated_data):
+        email = instance.email
+        with transaction.atomic():
+            instance = super(UserSerializer, self).update(instance, validated_data)
+            if validated_data.get('email') and email != instance.email and settings.get('SEND_ACTIVATION_EMAIL'):
```

Comment:
I'd swap the order of  these conditions to avoid doing redundant conditions
``` python
settings.get('SEND_ACTIVATION_EMAIL') and validated_data.get('email') and email != instance.email
```